http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/Boot.scala

```diff
@@ -47,119 +48,137 @@ import scala.concurrent.duration._
+          val rstudioDAO = new HttpRStudioDAO(appDependencies.clusterDnsCache)
+
+          implicit def clusterToolToToolDao =
+            ToolDAO.clusterToolToToolDao(jupyterDAO, appDependencies.welderDAO, rstudioDAO)
+
+          system.actorOf(
+            ClusterMonitorSupervisor.props(
+              monitorConfig,
+              dataprocConfig,
+              clusterBucketConfig,
+              appDependencies.googleDataprocDAO,
+              appDependencies.googleComputeDAO,
+              googleStorageDAO,
+              appDependencies.google2StorageDao,
+              appDependencies.dbReference,
+              authProvider,
+              autoFreezeConfig,
+              jupyterDAO,
+              rstudioDAO,
+              appDependencies.welderDAO,
+              leonardoService,
+              appDependencies.clusterHelper
+            )
+          )
+          system.actorOf(
+            ZombieClusterMonitor.props(zombieClusterMonitorConfig,
+                                       appDependencies.googleDataprocDAO,
+                                       googleProjectDAO,
+                                       appDependencies.dbReference)
           )
-        )
-        system.actorOf(
-          ZombieClusterMonitor.props(zombieClusterMonitorConfig,
+          system.actorOf(
+            ClusterToolMonitor.props(clusterToolMonitorConfig,
                                      appDependencies.googleDataprocDAO,
                                      googleProjectDAO,
-                                     appDependencies.dbReference)
-        )
-        system.actorOf(
-          ClusterToolMonitor.props(clusterToolMonitorConfig,
-                                   appDependencies.googleDataprocDAO,
-                                   googleProjectDAO,
-                                   appDependencies.dbReference,
-                                   appDependencies.metrics)
-        )
-      }
-      val clusterDateAccessedActor =
-        system.actorOf(ClusterDateAccessedActor.props(autoFreezeConfig, appDependencies.dbReference))
-      val proxyService = new ProxyService(proxyConfig,
-                                          appDependencies.googleDataprocDAO,
-                                          appDependencies.dbReference,
-                                          appDependencies.clusterDnsCache,
-                                          authProvider,
-                                          clusterDateAccessedActor)
-      val statusService = new StatusService(appDependencies.googleDataprocDAO,
-                                            appDependencies.samDAO,
+                                     appDependencies.dbReference,
+                                     appDependencies.metrics)
+          )
+
+          for {
+            _ <- appDependencies.clusterHelper.setupDataprocImageGoogleGroup()
+          } yield ()
+        }
```

Comment:
this IO is not chained to the main one...so I don't think this is actually doing anything.

So I would move this down to the main for comp, something like
```
for {
          _ <- if(leoExecutionModeConfig.backLeo) appDependencies.clusterHelper.setupDataprocImageGoogleGroup() else IO.unit
          _ <- IO.fromFuture {
            IO {
              Http()
                .bindAndHandle(leoRoutes.route, "0.0.0.0", 8080)
                .onError {
                  case t: Throwable =>
                    unsafeLogger.error(t)("FATAL - failure starting http server").unsafeToFuture()
                }
            }
          } >> IO.never
        } yield ()
```

Reply:
I'm not sure I understand how moving it makes a difference. Aren't we `flatMap`ing on `appDependencies.clusterHelper.setupDataprocImageGoogleGroup()` either way?

Reply:
P.S. My motivation for keeping it up inside the `if (backLeo)` block was to be able to see all backLeo-specific operations in one place.

Reply:
I think Qi's right. The expression:
```
for {
   _ <- appDependencies.clusterHelper.setupDataprocImageGoogleGroup()
} yield ()
```
itself returns an `IO`, which is never run or chained to any other IO. The `startup` method must return a single `IO`, which is run under the hood by `IOApp`. Therefore I agree with Qi's suggestion to chain together all the `IO`s in a single for-comp.

Reply:
Ah right...I forgot it's the `IOApp` that runs the returned `IO`.

Reply:
I get the compilation error

```
[error] Boot.scala:159:13: dead code following this construct
[error]           _ <- IO.fromFuture {
[error]             ^
[error] one error found
```

```scala
_ <- IO.fromFuture {
            IO {
              Http()
                .bindAndHandle(leoRoutes.route, "0.0.0.0", 8080)
                .onError {
                  case t: Throwable =>
                    unsafeLogger.error(t)("FATAL - failure starting http server").unsafeToFuture()
                }
            }
          } >> IO.never
```

I think it's because `IO.never` returns an `IO[Nothing]` but not sure. I have to look into this more.

Reply:
okay...in that case you probably need to move `IO.never` outside the for comp...
```
val res =   for {
          _ <- if(leoExecutionModeConfig.backLeo) appDependencies.clusterHelper.setupDataprocImageGoogleGroup() else IO.unit
          _ <- IO.fromFuture {
            IO {
              Http()
                .bindAndHandle(leoRoutes.route, "0.0.0.0", 8080)
                .onError {
                  case t: Throwable =>
                    unsafeLogger.error(t)("FATAL - failure starting http server").unsafeToFuture()
                }
            }
          }
        } yield ()

res >> IO.never
```

Reply:
now that I think about it...I think this kinda of one off environment specific setup should probably live in terraform just like leo's SA is created by terraform.

Reply:
Hmm...I can see a rationale for that. My reasons for not doing that would be:

1. From the ephemeral environment presentation, it sounded like account creation type of stuff is not automated in terraform. It'd be nice to not have to add to the 34 manual steps Greg mentioned during the presentation.
2. Even if we created the group elsewhere, we still need to do checks at boot time to make sure the group is there, etc. so we wouldn't be able to remove most of the code I added to Boot.
3. I think the code in Boot.scala is well factored and doesn't clobber the file.
4. We'd still need the config templating and make sure it matches the externally created group.
5. We already have the code in place that works.

Anyway...I've addressed the last code comments.

Reply:
I wasn't suggesting reverting the code in this PR since we really want this PR to be out..but more of thinking addressing it in a separate ticket.

> From the ephemeral environment presentation, it sounded like account creation type of stuff is not automated in terraform. It'd be nice to not have to add to the 34 manual steps Greg mentioned during the presentation.
I'm 90% sure we can automate this with terraform

> Even if we created the group elsewhere, we still need to do checks at boot time to make sure the group is there, etc. so we wouldn't be able to remove most of the code I added to Boot.
I'll buy the check argument.